src/Hangman/Puzzle.hs

```diff
@@ -32,28 +32,30 @@ stateAnnouncement puzzle =
         _ -> ""
 
 handleInput :: Puzzle -> String -> Puzzle
+handleInput puzzle [c] =
+    handleGuess puzzle c
 handleInput puzzle input =
-    case input of
-        [c] -> handleGuess puzzle c
-        _   -> puzzle { state = InvalidGuess input }
+    puzzle { state = InvalidGuess input }
 
 handleGuess :: Puzzle -> Char -> Puzzle
-handleGuess puzzle guess =
+handleGuess puzzle@(Puzzle _ completed' guesses' _) guess =
```

Comment:
I'd wondered what this syntax meant - looks pretty similar to object destructuring in JavaScript.